.github/workflows/test_e2e.yml

```diff
@@ -33,105 +41,100 @@ jobs:
-          echo "SKAFFOLD_DEFAULT_REPO=$SKAFFOLD_DEFAULT_REPO" >> "$GITHUB_ENV"
-          echo "CONTROL_IP=$CONTROL_IP" >> "$GITHUB_ENV"
-
-      - name: Build and Deploy HCCM
+      - uses: hetznercloud/tps-action@main
+        with:
+          token: ${{ secrets.HCLOUD_TOKEN  }}
+
+      - name: Setup environment
+        run: make -C dev up
+
+      - name: Run skaffold
         run: |
-          skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}"
-          tag=$(skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}" --quiet --output="{{ (index .Builds 0).Tag }}")
-          skaffold deploy --images=hetznercloud/hcloud-cloud-controller-manager=$tag
+          source dev/files/env.sh
+          skaffold run
 
       - name: Run tests
         run: |
+          source dev/files/env.sh
           go test ./tests/e2e -tags e2e -v -race -timeout 60m -coverprofile=coverage.txt
 
       - name: Upload coverage reports to Codecov
         uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
-      - name: Download logs & events
+      - name: Dump logs & events
         if: always()
         continue-on-error: true
         run: |
           mkdir debug-logs
+
+          echo "::group::hccm.log"
           kubectl logs \
             --namespace kube-system \
             --selector app.kubernetes.io/name=hcloud-cloud-controller-manager \
             --all-containers \
             --prefix=true \
             --tail=-1 \
-            > debug-logs/hccm.log
-          
+            | tee debug-logs/hccm.log
+          echo "::endgroup::"
+
+          echo "::group::events.yaml"
           kubectl get events \
             --all-namespaces \
             --sort-by=.firstTimestamp \
             --output yaml \
-          > debug-logs/events.yaml
-
-      - name: Show HCCM Logs on Failure
-        if: failure()
-        continue-on-error: true
-        run: |
-          echo "::group::hccm.log"
-          cat debug-logs/hccm.log
+          | tee debug-logs/events.yaml
           echo "::endgroup::"
 
-      - name: Cleanup test environment
+      - name: Cleanup
         if: always()
         continue-on-error: true
-        run: |
-          hack/dev-down.sh
+        run: make -C dev down
 
       - name: Persist debug artifacts
         if: always()
         continue-on-error: true
         uses: actions/upload-artifact@v4
         with:
-          name: debug-logs-${{ env.SCOPE }}
+          name: debug-logs-${{ env.ENV }}
           path: debug-logs/
 
   robot:
     name: Robot
+
     permissions:
       id-token: write
 
-    # Make sure that only one Job is using the server at a time
-    concurrency: robot-test-server
+    # Make sure that only one job is using the server at a time
+    concurrency: robot
     environment: e2e-robot
 
     env:
-      K3S_VERSION: v1.29.0+k3s1
-      SCOPE: gha-${{ github.run_id }}-${{ github.run_attempt }}-robot
+      K3S_CHANNEL: v1.29
```

Comment:
Should we add the renovate regex manager?

Reply:
To pin a specific version? We need to bump the channels (in non-robot matrix) & docs anyway when we upgrade, so I dont see the benefit here.